### PR TITLE
Pimoroni Tiny 2040 support

### PR DESCRIPTION
**Pull request by ptka**
## Summary
Added support for the Pimoroni Tiny 2040 device: boards/arm/rp2040/pimoroni-tiny2040

It’s an RP2040 based device with some interesting features:
 + small form factor (18x21.3mm)
 + 8 MByte Flash
 + RGB LED
 + Reset Button
 + USB-C
 - reduced pin-header

The support in derived from: boards/arm/rp2040/raspberrypi-pico

## Impact
Only file changed outside of board configuration folder: boards/Kconfig

## Testing
All included configuration tested on real device.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@ptka can you squash the patchset into one?

**Comment by ptka** [comment]
Can I still squash the open PR or do I need to close it and open a new one?

**Comment by pkarashchenko** [comment]
> Can I still squash the open PR or do I need to close it and open a new one? 

Yes, you still can sqash and then use `git push -f origin branch_name`

**Comment by ptka** [comment]
Thanks ... I managed to get it squashed.

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2cdev.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#ifndef __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2CDEV_H
+#define __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2CDEV_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <stdint.h>
```
Comment (pkarashchenko) [inline boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2cdev.h:29]:
```suggestion
 
#​ifndef​ __ASSEMBLY__
#  include <stdint.h>
#endif
```

File: boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2sdev.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#ifndef __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2SDEV_H
+#define __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2SDEV_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <stdint.h>
```
Comment (pkarashchenko) [inline boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2sdev.h:29]:
```suggestion
 
#​ifndef​ __ASSEMBLY__
#  include <stdint.h>
#endif
```

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+#include <nuttx/config.h>
+
+#include <debug.h>
+
+#include <nuttx/board.h>
+#include <arch/board/board.h>
+
+#include "arm_arch.h"
+#include "arm_internal.h"
```
Comment (pkarashchenko) [inline boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c:33]: Why do we need this include here?

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
```
Comment (pkarashchenko) [inline boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c:27]: Maybe we can replace it with `stdint.h`?

File: boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2cdev.h (hunk 1 above)
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2cdev.h:29]: Is this common practice in NuttX?

Comment (pkarashchenko) [reply boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2cdev.h]: You have the same in boards/arm/rp2040/pimoroni-tiny2040/include/board.h from you PR.
In general if you expect header files to be included in assembly then having `stdint.h` will definitely bring you into troubles.

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c (hunk 3 above)
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c:33]: It's part of the original code, this device support is based on.
Nevertheless ... let me check, if I can remove these includes.

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c (hunk 4 above)
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c:27]: Let me check this, too.

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c (hunk 3 above)
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_boardinitialize.c]: Includes removed.

File: boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c (hunk 4 above)
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/src/rp2040_gpio.c]: Include updated.

**Comment by ptka** [comment]
@pkarashchenko : PR updated and squashed with your suggestions. Thanks for review.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2sdev.h [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+#ifndef __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2SDEV_H
+#define __BOARDS_ARM_RP2040_PIMORONI_TINY2040_INCLUDE_RP2040_I2SDEV_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
```
[the remark this answers is not in the document]
Comment (ptka) [reply boards/arm/rp2040/pimoroni-tiny2040/include/rp2040_i2sdev.h]: Fixed :-)

**Pull request merged**